BLSQ/iaso
# POLIO-1333 vaccine stock management backend list view api

APIs and test of this APIs for the list view of Vaccine Stock Management Page.
Based on #964 

Related JIRA tickets : POLIO-1333, POLIO-1332, POLIO-1148, POLIO-1149

## Self proofreading checklist


## Changes

Add APIs for the list view of Vaccine Stock Management.
Make use of the models created in POLIO-1332 (See #964 )
Quite complex calculations in the API to calculate the `vials_received`, `vials_used`, `stock_of_usable_vials` ...


## How to test

See `plugins/polio/tests/test_vaccine_stock_management.py` for details 
But basically you need an `Account`, `DataSource`, `SourceVersion`, `Project` and a `OrgUnit` and its related `OrgUnitType` (COUNTRY).
Then a `Campaign` and a `Round`. (for the Same Country)
A `VaccineRequestForm` and related `VaccineArrivalReport`.
Then a `VaccineStock` object from the account, country, vaccine.
Then some `OutgoingStockMovement`, `DestructionReport` and `IncidentReport` to have enough data to calculate all the values.
(See in `setUpTestData` in the test).
